lib/primitives/mtx.js

```diff
@@ -1340,6 +1340,50 @@ class MTX extends TX {
     return select;
   }
 
+  /**
+   * Select and add coins to fill the transaction
+   * and set the output value to the total value minus fee
+   * @param {Coin[]} coins - list of coins to fill with
+   * @param {Object?} options - fill options
+   * @param {(Amount)?} options.threshold - highest coin value
+   * in satoshis that should be included in sweep
+   * @param {(Rate)?} options.rate - rate to pay for tx fee
+   * @returns {MTX} mtx - returns a filled mtx
+   */
+   async fill(coins, options) {
+    assert(coins.length, 'Must pass in coins to fill');
+    assert(options, 'Options are required');
+    assert(this.outputs.length, 'Must have at least 1 output');
+
+    // must set option to fill for coin selection
+    if (!options.fill)
+      options.fill = true;
+    // unless otherwise set, we want to sort and selct coins by value
+    // fill will automatically sort from lowest to highest
+    if (!options.select)
+      options.select = 'value';
+
+    // Select necessary coins.
+    const select = await this.selectCoins(coins, options);
+
+    // Add coins to transaction.
+    for (const coin of select.chosen)
+      this.addCoin(coin);
+    // Naive implentation for output and fees-
+    // Send all swept coins to first output set on mtx.
+    // Later can add more sophisticated sweeping to
+    // multiple addresses and subtractFee support
+    this.outputs[0].value = this.getInputValue() - select.fee;
```

Comment:
`tx` might not have output at all. Probably good idea to assert you we have output to modify. (at least until we have better output management for filling)

Reply:
This is asserted on line 1356. Expecting an output to be passed seems to be the expected behavior for `fund` too. The only difference here is that fund fills up to the total output value, where as here, the output value is whatever we reach after filling the tx.